ops/model.py

```diff
@@ -1484,6 +1518,14 @@ def remove_revision(self, revision: int):
         :class:`ops.SecretRemoveEvent <ops.charm.SecretRemoveEvent>` or
         :class:`ops.SecretExpiredEvent <ops.charm.SecretExpiredEvent>`.
 
+        If the charm does not have permission to remove the secret, or it no
+        longer exists, this method will succeed, but the unit will go into error
+        state on completion of the current Juju hook.
+
+        If the tracked revision is removed, then :meth:`get_content` should
+        be called immediately afterwards with ``refresh=True``, to avoid
+        leaving the unit in a 'stuck' state.
```

Comment:
I sorta don't think we need to say this. I think this is really a charm bug if they're going around removing the revision being tracked, and I'm thinking it just adds confusion here.

Reply:
My concern is that they don't know if they are doing this. For example, if there's an action "remove this revision" then the charm can't prevent against that being the tracked one, and it's tricky to figure out what's going wrong at that point.

What are the use-cases for `secret-remove` outside of rotation/expiry/remove-all? I read over the Secrets Core spec but couldn't find anything there. If we don't expect it to be used, instead of the above paragraph, could we strengthen the guidance/language, somewhat like:

```python
        """Remove the given secret revision.

        This is normally only called when handling
        :class:`ops.SecretRemoveEvent <ops.charm.SecretRemoveEvent>` or
        :class:`ops.SecretExpiredEvent <ops.charm.SecretExpiredEvent>`.

        Args:
            revision: The secret revision to remove. This should usually be set to
                :attr:`SecretRemoveEvent.revision` or :attr:`SecretExpiredEvent.revision`.
```

Reply:
Yeah, I think it's a good idea to add that. I can't see any reason why you'd `remove_revision` outside of those contexts. For example, why would there be an action "remove this revision" in the first place? That seems like a big charm code smell.